File: src/main/java/ar/edu/um/tesoreria/rest/model/Auditable.java
/**
 * 
 */
package ar.edu.um.tesoreria.rest.model;

import java.time.LocalDateTime;

import jakarta.persistence.Column;
import jakarta.persistence.EntityListeners;
import jakarta.persistence.MappedSuperclass;

import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.LastModifiedDate;
import org.springframework.data.jpa.domain.support.AuditingEntityListener;

import lombok.AccessLevel;
import lombok.Getter;
import lombok.Setter;

/**
 * @author daniel
 *
 */
@Getter(AccessLevel.PROTECTED)
@Setter(AccessLevel.PROTECTED)
@MappedSuperclass
@EntityListeners(AuditingEntityListener.class)
public class Auditable {
	@CreatedDate
	@Column(name = "created", nullable = false, updatable = false)
	private LocalDateTime created;
	
	@LastModifiedDate
	@Column(name = "updated")
	private LocalDateTime updated;
}

File: src/main/java/um/tesoreria/rest/kotlin/model/Auditable.kt
package um.tesoreria.rest.kotlin.model

import jakarta.persistence.Column
import jakarta.persistence.EntityListeners
import jakarta.persistence.MappedSuperclass
import org.springframework.data.annotation.CreatedDate
import org.springframework.data.annotation.LastModifiedDate
import org.springframework.data.jpa.domain.support.AuditingEntityListener
import java.time.LocalDateTime

@MappedSuperclass
@EntityListeners(AuditingEntityListener::class)
open class Auditable {

    @field:CreatedDate
    @Column(name = "created", nullable = false, updatable = false)
    open var created: LocalDateTime? = null

    @field:LastModifiedDate
    @Column(name = "updated")
    open var updated: LocalDateTime? = null
}
